### PR TITLE
Rover: fix guided mode for boat

### PR DESCRIPTION
**Pull request by tajisoft**
Stop vehicle with loiter for boat in GUIDED mode.
Loiter function is from ModeLoiter::udpate().
I tested it with SITL for FRAME_CLASS 1 and 2, it worked.
But I doesn't comfirm the liter behavior actually. Is there a way to comfirm than with SITL?

### CODE REVIEW

File: APMrover2/mode_guided.cpp [hunk 1]
```diff
@@ -147,3 +159,44 @@ void ModeGuided::set_desired_turn_rate_and_speed(float turn_rate_cds, float targ
...
+{
+    if (!rover.is_boat() || !_reached_destination) {
+        return;
+    }
+
+    // get distance (in meters) to destination
+    _distance_to_destination = get_distance(rover.current_loc, _destination);
+
+    // if within loiter radius slew desired speed towards zero and use existing desired heading
+    if (_distance_to_destination <= g2.loit_radius) {
```
Comment (muramura) [inline APMrover2/mode_guided.cpp]: I do not have any projects that abbreviate loiter as loit.
I think that loiter is good. Expressing the same word with multiple abbreviations does not make sense.

**Comment by muramura** [comment]
I think that the board will sail where there is a flow of water.
In this case it was mathematical problem to be conscious of the speed of the water and to depart toward the virtual point.
I think that it is better to incorporate this idea.

ddd

**Comment by tajisoft** [comment]
@rmackay9 Maybe I shouldn't force push for PR branch?

**Comment by rmackay9** [comment]
@tajisoft, it's fine to force push a PR branch.  txs

**Comment by tajisoft** [comment]
@rmackay9 I'm not sure it is a good way to access Loiter with rover.mode_loiter variable. But I think this is simplest way.

File: APMrover2/mode_guided.cpp [hunk 2]
```diff
@@ -48,7 +52,11 @@ void ModeGuided::update()
                 calc_steering_to_heading(_desired_yaw_cd);
                 calc_throttle(calc_reduced_speed_for_turn_or_distance(_desired_speed), true, true);
             } else {
-                stop_vehicle();
+                if (rover.is_boat()) {
+                    start_loiter();
```
Comment (peterbarker) [inline APMrover2/mode_guided.cpp]: What happens if `start_loiter` fails here?  What state does that leave us in?

Comment (rmackay9) [reply APMrover2/mode_guided.cpp]: start_loiter should probably return a bool to notify whether it could get into loiter or not.

Comment (tajisoft) [reply APMrover2/mode_guided.cpp]: @peterbarker It should be stopped when fail to get in Loiter. I fixed it.
@rmackay9 Thank you for your advice.

File: APMrover2/mode_guided.cpp [hunk 3]
```diff
@@ -31,8 +31,8 @@ void ModeGuided::update()
                 calc_steering_to_waypoint(_reached_destination ? rover.current_loc : _origin, _destination, _reversed);
                 calc_throttle(calc_reduced_speed_for_turn_or_distance(_reversed ? -_desired_speed : _desired_speed), true, true);
             } else {
-                if (rover.is_boat()) {
-                    start_loiter();
+                if (rover.is_boat() && start_loiter()) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Loiter in GUIDED mode");
```
Comment (rmackay9) [inline APMrover2/mode_guided.cpp]: Hi there.  I'd prefer if we don't send this text to the GCS.  Recently I'm pushing to only send messages when unusual things are happening or errors.  So when all is working OK, let's not send messages.

Comment (tajisoft) [reply APMrover2/mode_guided.cpp]: @rmackay9 Ok. I'll fix it. My idea was to let the user know that it was Loiter that it was stop_vehicle so far.

**Comment by tajisoft** [comment]
@rmackay9 I deleted 2nd condition and confirmed the operation with SITL.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**